packages/astro/src/cli/install-package.ts

```diff
@@ -0,0 +1,124 @@
+import boxen from 'boxen';
+import { execa } from 'execa';
+import { bold, cyan, dim, magenta } from 'kleur/colors';
+import { createRequire } from 'node:module';
+import ora from 'ora';
+import prompts from 'prompts';
+import whichPm from 'which-pm';
+import { debug, info, type LogOptions } from '../core/logger/core.js';
+
+type GetPackageOptions = {
+	skipAsk?: boolean;
+	cwd?: string;
+};
+
+export async function getPackage<T>(
+	packageName: string,
+	logging: LogOptions,
+	options: GetPackageOptions,
+	otherDeps: string[] = []
+): Promise<T | undefined> {
+	const require = createRequire(options.cwd ?? process.cwd());
+
+	let packageImport;
+	try {
+		require.resolve(packageName);
+
+		// The `require.resolve` is required as to avoid Node caching the failed `import`
+		packageImport = await import(packageName);
+	} catch (e) {
+		info(
+			logging,
+			'',
+			`To continue, Astro requires the following dependency to be installed: ${bold(packageName)}.`
+		);
+		const result = await installPackage([packageName, ...otherDeps], options, logging);
+
+		if (result) {
+			packageImport = await import(packageName);
+		} else {
+			return undefined;
+		}
+	}
+
+	return packageImport as T;
+}
+
+function getInstallCommand(packages: string[], packageManager: string) {
+	switch (packageManager) {
+		case 'npm':
+			return { pm: 'npm', command: 'install', flags: [], dependencies: packages };
+		case 'yarn':
+			return { pm: 'yarn', command: 'add', flags: [], dependencies: packages };
+		case 'pnpm':
+			return { pm: 'pnpm', command: 'add', flags: [], dependencies: packages };
+		default:
+			return null;
+	}
+}
+
+async function installPackage(
+	packageNames: string[],
+	options: GetPackageOptions,
+	logging: LogOptions
+): Promise<boolean> {
+	const cwd = options.cwd ?? process.cwd();
+	const packageManager = (await whichPm(cwd)).name ?? 'npm';
+	const installCommand = getInstallCommand(packageNames, packageManager);
+
+	if (!installCommand) {
+		return false;
+	}
+
+	const coloredOutput = `${bold(installCommand.pm)} ${installCommand.command}${[
+		'',
+		...installCommand.flags,
+	].join(' ')} ${cyan(installCommand.dependencies.join(' '))}`;
+	const message = `\n${boxen(coloredOutput, {
+		margin: 0.5,
+		padding: 0.5,
+		borderStyle: 'round',
+	})}\n`;
+	info(
+		logging,
+		null,
+		`\n  ${magenta('Astro will run the following command:')}\n  ${dim(
```

Comment:
Should `message` be rendered after "...following command:"?

Reply:
I also thought that was weird while doing the code, but in the terminal it makes a bit more sense visually. This is also how it works for `astro add` (code was copied from there)